### ISSUE TITLE
Integration with Dataloader

### ISSUE THREAD

**Issue by dolfinus** [absinthe-graphql/absinthe_relay#168]
Hi.

I've tried to use both absinthe-relay (for pagination) and dataloader (for handling N+1 problem) in the same project.
But I haven't found some working way to combine them in the documentatio.

So I've impletented a simple helper module which allows to integrate these libraries wich each other:
https://github.com/dolfinus/arkenston-backend/blob/6a464931d75e01cfd7ee482dcf3315666933a563/lib/arkenston_web/schema/helpers/pagination.ex#L15
https://github.com/dolfinus/arkenston-backend/blob/6a464931d75e01cfd7ee482dcf3315666933a563/lib/arkenston/helper/query_helper.ex#L233

Now I can use `paginated` macros:
https://github.com/dolfinus/arkenston-backend/blob/6a464931d75e01cfd7ee482dcf3315666933a563/lib/arkenston_web/schema/helpers/revision.ex#L34.

How about to include such a helper to absinthe-relay? IMHO, I'll be helpful.

**Comment by benwilson512** [comment]
Hey @dolfinus, thanks for these helpers. Until recently, this kind of integration wasn't possible because there wasn't an easy way for Ecto (and by extension dataloader) do unique limits for each parent. However now that Dataloader does support limits, this sort of thing can re-evaluated again.

**Comment by HangingClowns** [comment]
hi @benwilson512 circling back from our conversation on slack. I'm trying to think about how we can do this in a good way. Here's my idea:

`import Absinthe.Relay.Resolution.Helpers, only: [dataloader: 1]`

then for the type:
```elixir
object :parent do
    field :id, :integer
    field :child, resolve: dataloader(Module, order: [desc: :inserted_at])
end
```

I think the most difficult part is trying to get the sorting correct, so we may consider the above OR maybe rely on the dataloader source to specify the sort?

**Comment by woylie** [comment]
I'm using both Absinthe Relay and Dataloader in a project as well, but haven't used Dataloader for Relay connection fields yet. In my case, I'd require the sorting to be handled by the query function of the Dataloader source, without the need to specify the order as in the code example above (potentially taking the sort field/order from query arguments).

**Comment by dolfinus** [comment]
IMHO, sorting should be handled by a query function, because some of users could want to pass ordering as an argument. There should be no hard coded values.

**Comment by benwilson512** [comment]
Agreed about sorting. @HangingClowns I guess I'm a bit confused by your example snippet. That code works today. The "integration with dataloader" part of this issue has to do with combining connection fields with dataloader. Using dataloader on node entities works just fine already.

**Comment by HangingClowns** [comment]
hi @benwilson512 from what I understand, paginating the data using relay doesnt' work? hence i was thinking we can do this: `import Absinthe.Relay.Resolution.Helpers, only: [dataloader: 1]`, notice how it's importing from Absinthe.Relay INSTEAD of just Absinthe. Maybe the name is too ambiguous, though?
